### PR TITLE
feat: Migrate all pool types to new join/exit flow

### PR DESCRIPTION
**Pull request by alter-eggo**
# Description

Migrates all pool types to use the join/exit providers and their associated services and handlers. Following this PR all the old join/exit logic should be redundant.

Notes:
- For ComposableStableV1 pools we need to use generalized exits because the pool type doesn't support the proportional exit type.

Original work was done here: https://github.com/balancer/frontend-v2/pull/2690

## Type of change

- [x] Code refactor / cleanup

## How should this be tested?

- [ ] Test all types of join/exit transactions for all the different pool types.

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have requested at least 1 review (If the PR is significant enough, use best judgement here)
- [ ] I have commented my code where relevant, particularly in hard-to-understand areas
- [x] If package-lock.json has changes, it was intentional.
- [x] The base of this PR is `master` if hotfix, `develop` if not

### CODE REVIEW

File: src/providers/local/join-pool.provider.ts [hunk 1]
```diff
@@ -309,10 +324,12 @@ export const joinPoolProvider = (
   async function join(): Promise<TransactionResponse> {
     try {
       txError.value = '';
-      joinPoolService.setJoinHandler(isSingleAssetJoin.value);
+
+      joinPoolService.setJoinHandler(joinHandlerType.value);
       setApprovalActions();
 
-      return joinPoolService.join({
+      console.log('joinHandler:', joinHandlerType.value);
```
Comment (agualis) [inline src/providers/local/join-pool.provider.ts:331]: @garethfuller Forgot removing this one?

File: src/providers/local/exit-pool.provider.ts [hunk 2]
```diff
@@ -342,6 +350,7 @@ export const exitPoolProvider = (pool: Ref<Pool>) => {
 
     exitPoolService.setExitHandler(exitHandlerType.value);
 
+    console.log('exitHandler:', exitHandlerType.value);
```
[the remark this answers is not in the document]
Comment (garethfuller) [reply src/providers/local/exit-pool.provider.ts:353]: Yup

**Review by agualis — APPROVED** [review]

File: src/components/contextual/pages/pool/add-liquidity/composables/useMyWallet.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  const { pool, isLoadingPool } = usePool();
+  const { setAmountsIn, setJoinWithNativeAsset, isSingleAssetJoin, amountsIn } =
+    useJoinPool();
+  const { nativeAsset, wrappedNativeAsset, getMaxBalanceFor } = useTokens();
+  const { activeTab } = useInvestPageTabs();
+
+  /**
+   * COMPUTED
+   */
+  const excludedTokens = computed<string[]>(() => {
```
Comment (alter-eggo) [inline src/components/contextual/pages/pool/add-liquidity/composables/useMyWallet.ts:24]: don't understand this, why we put pool address to `excluded tokens`, maybe problem in naming here?

File: src/pages/_layouts/FocussedLayout.vue [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<script setup lang="ts">
+import FocussedLayout from '@/components/layouts/FocussedLayout.vue';
+</script>
+
+<template>
+  <FocussedLayout>
```
Comment (alter-eggo) [inline src/pages/_layouts/FocussedLayout.vue:6]: why they have same names? maybe name component as `CFocussedLayout` (C - refers to 'component')?

Comment (alter-eggo) [reply src/pages/_layouts/FocussedLayout.vue:6]: its always a pain, when there are vue files with same name

**Review by alter-eggo — COMMENTED** [review]
great work guys 💪 @agualis @garethfuller

**Pull request merged**